### PR TITLE
Add mailing list attribute to project schema

### LINKED ISSUE #701: Add mailing list attribute to project schema
**Describe the solution you'd like**
  - Extend [project](https://github.com/OWASP/Nest/blob/9fd5d355120cf7474777b5b2c4db1357746f9681/schema/project.json#L32) schema with an optional mail list attribute. 
  - pick best of different options for the name (maillist, mailing_list, mail_list, etc)
  - extend schema
  - add/extend tests for the new attribute

### PR DESCRIPTION
**Pull request by Raghavg18**
Resolves #701

### CODE REVIEW

File: schema/tests/data/project/negative/mail-invalid.yaml
Comment (AbhayTopno) [inline schema/tests/data/project/negative/mail-invalid.yaml:1]: rename this file and name it mail-none.yaml

File: schema/tests/project_test.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ def test_positive(project_schema):
             "license-invalid-value.yaml",
             "'INVALID-LICENSE-VALUE' is not one of ['AGPL-3.0', 'Apache-2.0', 'BSD-2-Clause', 'BSD-3-Clause', 'CC-BY-4.0', 'CC-BY-SA-4.0', 'CC0-1.0', 'EUPL-1.2', 'GPL-2.0', 'GPL-3.0', 'LGPL-2.1', 'LGPL-3.0', 'MIT', 'MPL-2.0', 'OTHER']",
         ),
+        ("mailing_list-invalid.yaml", "'invalid-email' is not a valid email format"),
```
Comment (AbhayTopno) [inline schema/tests/project_test.py:67]: Replace this line of code with this ` ("mailing_list-none.yaml", "None is not of type 'string'"), `

File: schema/tests/data/project/negative/mail-invalid.yaml
Comment (AbhayTopno) [inline schema/tests/data/project/negative/mail-invalid.yaml:1]: replace this file name with ` mailing_list-none.yaml `

File: schema/tests/data/project/negative/mail-empty.yaml
Comment (AbhayTopno) [inline schema/tests/data/project/negative/mail-empty.yaml:1]: replace the file name with ` mailing_list-empty.yaml `

File: schema/tests/data/project/negative/mail-empty.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+audience: builder
+leaders:
+  - github: leader-name-1
+    name: Leader Name 1
+  - github: leader-name-2
+    name: Leader Name 2
+level: 2
```
Comment (AbhayTopno) [inline schema/tests/data/project/negative/mail-empty.yaml:7]: after this line add ` mailing_list: "" `

File: schema/tests/data/project/negative/mail-invalid.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+audience: builder
+leaders:
+  - github: leader-name-1
+    name: Leader Name 1
+  - github: leader-name-2
+    name: Leader Name 2
+level: 2
```
Comment (AbhayTopno) [inline schema/tests/data/project/negative/mail-invalid.yaml:7]: after this line add ` mailing_list: `

**Review by AbhayTopno — COMMENTED** [review]
@Raghavg18 add this changes and it should work

**Comment by AbhayTopno** [comment]
And also add test case in ` schema/tests/data/project/positive/mail-empty.yaml `

**Review by arkid15r — CHANGES_REQUESTED** [review]
Could you remove unrelated changes and make sure tests work before requesting review?

**Comment by Raghavg18** [comment]
The test was not running even before this changes ig @arkid15r

**Comment by arkid15r** [comment]
> The test was not running even before this changes ig @arkid15r

I'm not sure what you mean. In any case your PR (as any other PR) must have all checks passed, including the tests.

**Comment by YourBroCode** [comment]
@Raghavg18 you can run Schema-test locally before making a PR:
- install pytest in your local system then move to  the folder of project_test.py file.
- run :  pytest project_test.py

**Comment by Raghavg18** [comment]
@arkid15r there are many conflicts in this branch so i have to push the code in new branch so i have to raise a new PR for this issue
